com.unity.ml-agents/Runtime/Sensors/RenderTextureSensorComponent.cs

```diff
@@ -102,5 +104,17 @@ internal void UpdateSensor()
                 m_Sensor.CompressionType = m_Compression;
             }
         }
+
+        /// <summary>
+        /// Clean up the sensor created by CreateSensors().
+        /// </summary>
+        public void Dispose()
+        {
+            if (!ReferenceEquals(null, m_Sensor))
+            {
+                m_Sensor.Dispose();
```

Comment:
could this be a double dispose?  looks safe since the sensor checks for null.  Just curious if it's actually needed or not.

Reply:
I made sure all the sensor.Dispose's are idempotent - they check for null textures first, then set the texture to null.